spec/account/account-manager.spec.js

```diff
@@ -57,69 +38,114 @@ describe("AccountManager", () => {
                 }
             }
         };
-        accountManager = new AccountManager(accountServiceFake, config, hotpFake, otDateMock, randomMock, promiseMock);
+        hotpFake = jasmine.createSpyObj("hotpFake", [
+            "generateSecretAsync",
+            "verifyToken"
+        ]);
+        hotpFake.generateSecretAsync.andCallFake(() => {
+            return promiseMock.resolve("thisisasecrectcodefrommfa");
+        });
+        hotpFake.verifyToken.andCallFake((key, token, opts) => {
+            return token !== "987654";
+        });
+        otDateMock = require("../mock/ot-date-mock");
+        promiseMock = require("../mock/promise-mock");
+        randomMock = require("../mock/random-mock");
+        accountManager = require("../../lib/account/account-manager")(accountServiceFake, config, hotpFake, otDateMock, promiseMock, randomMock);
     });
-    describe(".initiate()", () => {
-        it("gets an object back from service", (done) => {
-            accountManager.initiateAsync({
+    describe(".signupInitiationAsync()", () => {
+        it("gets the registration id back", (done) => {
+            accountManager.signupInitiationAsync({
                 email: "some.one@example.net"
             }).then((result) => {
                 expect(result).toEqual({
-                    accountId: jasmine.any(Buffer),
-                    email: "some.one@example.net",
-                    mfaKey: "thisisasecrectcodefrommfa",
-                    salt: jasmine.any(Buffer)
+                    regId: jasmine.any(String)
                 });
-                expect(result.accountId.length).toBe(24);
-                expect(result.salt.length).toBe(128);
+                expect(result.regId.length).toBe(24);
+            }).then(done, done);
+        });
+        it("gets the registration id back using config options", (done) => {
+            var accountManagerLocal;
 
+            accountManagerLocal = require("../../lib/account/account-manager")(accountServiceFake, {
```

Comment:
You are changing config options and making a lot of lines duplicated in your test.  Instead of exposing a bunch of mocks, why not expose a function to make the manager?  You can put this in your beforeEach().

```
create = (config) => {
    return require("../../lib/account/account-manager")(accountServiceFake, config || defaultConfig, hotpFake, otDateMock, promiseMock, randomMock);
}
```

Remove the creation of accountManager in the beforeEach().  Then just call `accountManager = create({...})` in your tests.